Orange/widgets/visualize/owmosaic.py

```diff
@@ -447,6 +448,9 @@ def updateGraph(self, data=-1, subsetData=-1, attrList=-1, **args):
         if data == -1:
             data = self.data
 
+        if any(attr.is_continuous for attr in data.domain):
+            data = Discretize(method=EqualFreq(n=4))(data)
```

Comment:
Here's a better way: instead of using a local variable `data`, use 

    if any(attr.is_continuous for attr in data.domain):
        self.discrete_data = Discretized(method=EqualFreq(n=4))(data)
    else:
        self.discrete_data = self.data

With this, you can avoid additional argument in `self.DrawData` - it would just use `self.discrete_data`. Also, you don't need re-discretize the data in `sendSelectedData`.